File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/response/HttpResponse.java
package com.ashlikun.okhttputils.http.response;

import com.ashlikun.okhttputils.http.HttpUtils;
import com.google.gson.annotations.SerializedName;

/**
 * 作者　　: 李坤
 * 创建时间: 10:27 admin
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：http的基本类
 */
public class HttpResponse extends AbsHttpResponse {
    public static int SUCCEED = 1;//正常请求
    public static int ERROR = 0;//请求出错
    public final static String CODE_KEY = "code";
    public final static String MES_KEY = "msg";

    @SerializedName(value = CODE_KEY)
    public int code = ERROR;
    @SerializedName(MES_KEY)
    public String message;

    public HttpResponse() {
        super();
    }

    public HttpResponse(String json) {
        super(json);
    }

    @Override
    public void setOnGsonErrorData(String json) {
        this.code = getIntValue(HttpResponse.CODE_KEY);
        this.message = getStringValue(HttpResponse.MES_KEY);
    }

    @Override
    public int getCode() {
        return code;
    }

    @Override
    public void setCode(int code) {
        this.code = code;
    }


    @Override
    public String getMessage() {
        return HttpUtils.unicode2String(message);
    }

    @Override
    public void setMessage(String message) {
        this.message = message;
    }

    /**
     * 是否成功  success
     */
    @Override
    public boolean isSucceed() {
        return code == SUCCEED;
    }
}

File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/response/HttpResponse.kt
package com.ashlikun.okhttputils.http.response

import com.google.gson.annotations.SerializedName

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.17 9:34
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：http的基本类
 */
open class HttpResponse(
    //原始数据
    json: String = ""
) : AbsHttpResponse(json) {
    @SerializedName(value = CODE_KEY, alternate = [CODE_KEY2])
    override var code = ERROR

    @SerializedName(MES_KEY, alternate = [MES_KEY2])
    override var message: String = ""

    override fun setOnGsonErrorData(json: String) {
        super.setOnGsonErrorData(json)
        code = getIntValue(CODE_KEY)
        message = getStringValue(MES_KEY)
    }

    /**
     * 是否成功  success
     */
    override val isSucceed: Boolean
        get() = code == SUCCEED

    companion object {
        var SUCCEED = 1 //正常请求
        var ERROR = 0 //请求出错
        const val CODE_KEY = "code"
        const val CODE_KEY2 = "status"
        const val MES_KEY = "msg"
        const val MES_KEY2 = "message"
    }
}